### DISCUSSION TITLE
Backwards Incompatible Changes

### DISCUSSION THREAD

**Discussion by chutcheson** [racket/rhombus#184] [General]
I was reading through the motivation document and I was really curious:

> make backwards-incompatible changes that reflect newer thinking (e.g., structs)

What kinds of backwards-incompatible changes would be beneficial that reflect newer thinking? Also, I'm really curious: what improvements have been made to structs?

**Comment by rocketnia** [comment]
You might be able to find little examples all over the place in the issue tracker. I'm not sure where to begin. Most Rhombus proposals have a little bit of backwards incompatibility to them in some sense. If they didn't, they could just be Racket proposals.

(To be fair, some of them are long-standing Racket wish list items that people are surfacing as possible out-of-the-gate features for Rhombus. Some of them may be ideas that people proposed for Rhombus first, perhaps having not even thought of suggesting them for Racket. Some of them are synergistic combinations of multiple ideas, which may collectively fall somewhere in between.)

Rhombus is primarily being thought of as a second `#lang` that would coexist alongside `#lang racket`. It can be "backwards incompatible" in the sense of harboring a different set of habits, core syntaxes, and terminology than what `#lang racket` programmers would use, but most ideas people have proposed for Rhombus would cause no breakage to `#lang racket` programs, and keeping `#lang racket` usable is an oft-stated goal.

As far as `struct` goes in particular, these are some of the ideas on people's minds that might culminate in a new meaning of `struct` in Rhombus:

* #57
* #76
* #89
* #92
* #152
